### DISCUSSION TITLE
Sterowanie roletami

### DISCUSSION THREAD

**Discussion by Kirkov24** [lkankowski/arduino-multi-relay2#41] [Ideas]
Czy jest możliwość ustawienia przycisków tak by ewentualnie można nimi wysterować przekaźniki zewnętrznych rolet? Chodzi oczywiście o programowe wykluczenie możliwości podania napięcia przez dwa przekaźniki które mają sterować tym samym silnikiem rolety. 

BTW: z uwagi na dużą ilość rolet sterowanie roletami mógłbym przenieść do oddzielnego hardwaru, zgadzam się z autorem, że główna funkcjonalność jego rozwiązania to sterowanie oświetleniem i dla bezpieczeństwa może nie warto dokładać dodatkowych funkcjonalności które mogłyby by wprowadzać błędy...

**Comment by lkankowski** [comment]
Nie miałem do czynienia z roletami, więc dla upewnienia napiszę jak to rozumiem:
1. roleta ma 2 kierunki, więc zapewne kierunek podaje się osobno 2 przewodami (nie licząc masy lub common)
2. zatem chodzi o to, aby na 2 przekaźnikach nie podać zasilania dla tej samej rolety

Jeżeli dobrze rozumiem, to nie ma obecnie takiej funkcjonalności i nie ma tego w najbliższych planach.
W sumie ciekawy pomysł na rozbudowę i mieściło by się w zakresie funkcjonalności tego projektu. Niestety, obecnie nie mam czasu na rozwój, bo w 100% zaspokaja moje potrzeby, a nie mam rolet. Chętnie przyjmę kontrybucję, pomogę podczas przeglądu kodu i dołączę do repozytorium, jeżeli spełni wymogi jakościowe.

Gdzieś obiła mi się o uszy nazwa "Roller shutter" (chyba https://github.com/gryzli133/RollerShutterSplit), więc zapewne jest to coś, co dobrze robi swoją robotę.

**Reply by pepsonEL** [reply]
Ja używam od 2 lat tego kodu co wskazałes do rolet i śmiga super. A Twojego do oświetlenia i tak samo zero problemów. Jednak chce powoli przejść na jakiś projekt oparty o ESP32 aby używać go w HA z ESPHome. Jednak na razie szukam. Jest boneIO ESP ale cena zabija mnie i tam sumarycznie na rolety i oświetlenie mieć aż ich 4 sterowniki co cenowo to kosmos.
Obecnie bardziej potrzebuje ESP32 z portem LAN i ekspnderami bo przekaźniki SSR mam więc bym je użył.

**Reply by lkankowski** [reply]
No i kwestia, o której zapewne pisałem w różnych miejscach - im mniej odpowiedzialności sterownika, tym lepiej, bo mniej okazji do awarii. W razie awarii padnie tylko oświetlenie, albo tylko rolety.

**Comment by leesiuu** [comment]
Odkopię trochę temat, bo może rzucę pomysłem, który jest łatwy w realizacji, a spełni te wymagania, o których jest tu wspomniane.
Mianowicie teraz przy konfiguracji przekaźnika mamy opcję podania pinu dependOn - ten zależny przekaźnik musi się włączyć, żeby zadziałał głowny przekaźnik.
Coś takiego tylko zależny przekaźnik musi być wyłączony? Ewentualnie flaga dla dependOn w konfiguracji, co stylu:
RELAY_DEPEND_ON_TURN_ON
RELAY_DEPEND_ON_TURN_OFF

Osobiście nie jestem na tym poziomie zaawansowania w Arduino, żeby taką funkcję zrealizować i dodać do sketcha do oceny, dlatego też mogę tylko tyle dopowiedziec :)
